pkg/initializers/types/types.py

```diff
@@ -10,13 +10,33 @@ class HuggingFaceDatasetInitializer:
     access_token: Optional[str] = None
 
 
+# Configuration for the S3 dataset initializer.
+@dataclass
+class S3DatasetInitializer:
+    storage_uri: str
+    endpoint: Optional[str] = None
+    access_key_id: Optional[str] = None
+    secret_access_key: Optional[str] = None
```

Comment:
@rudeigerc @akshaychitneni Does OpenDAL support IAM roles for IRSA ?
Like in cache: https://github.com/kubeflow/trainer/blob/master/pkg/initializers/types/types.py#L26C5-L26C13